File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Gambia.m
//
//  UIImage+Gambia.m
//  CountriesFlags
//

#import "Countries/UIImage+Gambia.h"

@implementation UIImage (Gambia)

+ (UIImage * _Nullable)flagGambiaWithSize:(CGSize)size {
    UIColor* colorGreen = [UIColor colorWithRed:28.0f/255.0f
                                          green:119.0f/255.0f
                                           blue:40.0f/255.0f
                                          alpha:1.0f];
    UIColor* colorRed = [UIColor colorWithRed:206.0f/255.0f
                                           green:17.0f/255.0f
                                            blue:38.0f/255.0f
                                           alpha:1.0f];
    UIColor* colorBlue = [UIColor colorWithRed:12.0f/255.0f
                                         green:28.0f/255.0f
                                          blue:140.0f/255.0f
                                         alpha:1.0f];
    UIColor* colorWhite = [UIColor whiteColor];
    
    UIGraphicsBeginImageContextWithOptions(size, YES, 0.0f);
    
    [colorRed setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorWhite setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 3.0f, size.width, size.height / 3.0f));
    
    [colorGreen setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 3.0f * 2.0f, size.width, size.height / 3.0f));
    
    [colorBlue setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 18.0f * 7.0f, size.width, size.height / 18.0f * 4.0f));
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Gambia.swift
//
//  UIImage+Gambia.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Gambia"wikipedia"
     
     Generates the flag of Gambia.
     For more information see [wikipedia].

     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Gambia.
     */
    internal static func flagGambia(with size: CGSize) -> UIImage? {
        let colorGreen = UIColor(red: 28.0/255.0,
                                 green: 119.0/255.0,
                                 blue: 40.0/255.0,
                                 alpha: 1.0)
        let colorRed = UIColor(red: 206.0/255.0,
                               green: 17.0/255.0,
                               blue: 38.0/255.0,
                               alpha: 1.0)
        let colorBlue = UIColor(red: 12.0/255.0,
                                green: 28.0/255.0,
                                blue: 140.0/255.0,
                                alpha: 1.0)
        let colorWhite = UIColor.white

        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)

        colorRed.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorWhite.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0, y: size.height / 3.0),
                          size: CGSize(width: size.width, height: size.height / 3.0)))

        colorGreen.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0, y: size.height / 3.0 * 2.0),
                          size: CGSize(width: size.width, height: size.height / 3.0)))

        colorBlue.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0, y: size.height / 18.0 * 7.0),
                          size: CGSize(width: size.width, height: size.height / 18.0 * 4.0)))

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }
}

#endif
